frontend/src/utilities/notebookControllerUtils.ts

```diff
@@ -86,15 +87,15 @@ export const verifyEnvVars = async (
   namespace: string,
   kind: string,
   envVars: Record<string, string>,
-  fetchFunc: (resourceName: string) => Promise<EnvVarResource>,
-  createFunc: (resource: Secret | ConfigMap) => Promise<EnvVarResource>,
-  replaceFunc: (resourceName: string, resource: EnvVarResource) => Promise<EnvVarResource>,
-  deleteFunc: (resourceName: string) => Promise<DeleteStatus>,
+  fetchFunc: (projectName: string, resourceName: string) => Promise<EnvVarResource>,
+  createFunc: (resource: EnvVarResource) => Promise<EnvVarResource>,
+  replaceFunc: (resource: EnvVarResource) => Promise<EnvVarResource>,
+  deleteFunc: (projectName: string, resourceName: string) => Promise<DeleteStatus>,
```

Comment:
It occurred to me just now, we should probably just do something like:

```
verifyEnvVars = async <T extends K8sResourceCommon>(
  ...,
  fetchFunc: ResourceGetter<T>,
  createFunc: ResourceCreator<T>,
  ...,
)
```
That way it's more generic and we don't need this `EnvVarResource` thing that we have to keep updating for any resource we want to do. It's a signature structure to getting, creating, updating, and removing a resource.

Also don't forget to make the method description generic -- it states Secret & ConfigMap.